### ISSUE TITLE
Make ++ an n-ary operator

### ISSUE THREAD

**Issue by CameronBieganek** [tlienart/PlusPlus.jl#1]
I propose that the code be changed to this:

```julia
module PlusPlus

export (++)

++(args::Union{AbstractString,AbstractChar}...) = *(args...)

++(args...) = vcat(args...)

end
```

This takes advantage of the fact that `++` is parsed as an _n_-ary operator. So now `'a' ++ 'b' ++ 'c'` won't throw a method error.

EDIT: I also removed the `(++)(x::AbstractVector, y::AbstractVector) = vcat(x, y)` method, since the generic fallback works fine for vectors, along with matrices and data frames.

**Comment by tlienart** [comment]
Sure :-) I’ll add you as collaborator, do what you think is best!

**Comment by tlienart** [comment]
(and add yourself on the License)

**Comment by tlienart** [comment]
ok well I did what you suggested + added you on the License :)

**Comment by CameronBieganek** [comment]
Thanks!

**Comment by tlienart** [comment]
Ah it’s a bit annoying, the package registration takes 3 days so I can’t tag 0.2

**Comment by waldyrious** [comment]
> `@tlienart` closed this 21 hours ago

For future reference, this was implemented in commit [6d1767c](https://github.com/tlienart/PlusPlus.jl/commit/6d1767c80dca192aae2926d4dea339bd49cbc06f#diff-fa49aadae08d5a4019b9d2d29e302371).
